### ISSUE TITLE
Difficulty in Special_yv regression?

### ISSUE THREAD

**Issue by DavidEnriqueNieves** [yu-rp/KANbeFair#10]
Good morning,

I've been having issues with Special_yv while working on training, and I've noticed that neither KAN nor MLPs do well when trying to perform regression with it. For example, using this command:

` python3 ./train.py --dataset Special_yv  --model KAN --kan_bspline_order 5 --epochs 500 --seed 1314 --layers_width 5  --loss mse --kan_bspline_grid 20`

I get a stagnant Average Training Loss of ~389, and Testing Loss of ~889, and I'm not sure why. It decreases slightly, reaching ~386 and 885, but it doesn't get anywhere close to a good value. 

I also got similar results attempting to use the MLP for it. I'm mainly posting to see if you have encountered similar difficulties.

**Comment by yu-rp** [comment]
Hi @DavidEnriqueNieves , thank you for the issue.

This issue primarily arises due to the inherent properties of the `special.yvp` function. Near 0, the values of the `special.yvp` function diverge and can become infinitely large. Consequently, some data samples $(x, y)$ have $y$ values that are very large, resulting in an extremely large MSE loss and gradients, which hinders the convergence of the algorithm.

A feasible approach is to employ a normalization method to adjust the range of data values before feeding them into the neural network. Another method is to avoid regions with extremely large values. For instance, when the first parameter (order) of `special.yvp` is 1, setting the second parameter to be larger than 0.5 can prevent the occurrence of the large loss values you mentioned.

You may also refer to [this page](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.yvp.html#scipy.special.yvp), which includes a graph at the end that illustrates the function behavior of `special.yvp`. This can help you intuitively understand the range of values that the `special.yvp` function can take.

**Comment by DavidEnriqueNieves** [comment]
Sounds good. I guess it makes sense as to why it wasn't included in the comparison. I do wonder what regions and/or preprocessing techniques were employed  in the original KAN paper. I think this should be fine for now.
